sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnostics.java

```diff
@@ -189,6 +189,25 @@ public Set<String> getContactedRegionNames() {
         return this.clientSideRequestStatistics.getContactedRegionNames();
     }
 
+    /**
+     * Gets the UserAgent header value used by the client issueing this operation
+     * @return the UserAgent header value used for the client that issued this operation
+     */
+    public String getUserAgent() {
+        String userAgent = null;
+        if (this.feedResponseDiagnostics != null) {
+            userAgent = this.feedResponseDiagnostics.getUserAgent();
+        } else if (this.clientSideRequestStatistics != null) {
+            userAgent = this.clientSideRequestStatistics.getUserAgent();
+        }
+
+        if (userAgent != null) {
```

Comment:
Can this ever be null? From the current code, it doesn't look like so.

Reply:
You are right. Fixed.